include/grpcpp/impl/codegen/method_handler_impl.h

```diff
@@ -59,33 +59,46 @@ class RpcMethodHandler : public MethodHandler {
       : func_(func), service_(service) {}
 
   void RunHandler(const HandlerParameter& param) final {
-    RequestType req;
-    Status status = SerializationTraits<RequestType>::Deserialize(
-        param.request.bbuf_ptr(), &req);
     ResponseType rsp;
+    Status status = param.status;
     if (status.ok()) {
-      status = CatchingFunctionHandler([this, &param, &req, &rsp] {
-        return func_(service_, param.server_context, &req, &rsp);
+      status = CatchingFunctionHandler([this, &param, &rsp] {
+        return func_(service_, param.server_context,
+                     static_cast<RequestType*>(param.request), &rsp);
```

Comment:
I'm legitimately bother by the use of `void*` and static upcasts like this in this file and the next one. I have to believe that there is a better option. I know that this is deep in internals for now, but I am concerned about whether or not this will actually work in all our use cases.

Additionally, adding an extra required allocation/deallocation (even if there are no interceptors) in all the code pathways is problematic. Maybe not so much for sync API which is not performance-optimized, but more for callback API (server-side) which does aim for performance.

Reply:
I think we really need to revisit this issue carefully.